### DISCUSSION TITLE
Using alternative code generators (namely LLVM and JS)

### DISCUSSION THREAD

**Discussion by ghost** [srid/haskell-flake#322] [General]
Is it possible you could provide a minimal example for different backends (documented here: https://downloads.haskell.org/~ghc/9.10.1/docs/users_guide/codegens.html)? I believe there is support for WASM output now as well... even sparse guidance in setting me down the right path would be useful.

I'll add that I'm using the package sets and devtools manged by Horizon (https://gitlab.horizon-haskell.net/package-sets). This tends to get updated a bit quicker.

**Comment by roberth** [comment]
I believe the way to do it is to have multiple projects, each with the right [`basePackages`](https://flake.parts/options/haskell-flake#opt-perSystem.haskellProjects._name_.basePackages) for the target you want. So essentially projects will correspond to package sets.
IIUC the Horizon sets are shaped like the Nixpkgs ones, so that should be fine.
I haven't done this myself so I don't have an example; would be good to have one.

**Reply by ghost** [reply]
With the right package set (still need to figure this out for the different targets), I think compiler flags still need to be set/replaced… I recall seeing GCC flags with the minimal example. Not quite clear on the full picture beyond setting the right basePackages.

**Reply by srid** [reply]
https://community.flake.parts/haskell-flake/settings

Mayee [these options](https://github.com/srid/haskell-flake/blob/5a86b9d5a1ad01e46d44af5fdaea59555b1199ca/nix/modules/project/settings/all.nix#L176-L189) are relevant.

---

Once you figure this out, @riskygit, it would be great to have something in the haskell-flkake documentation for this, especially using the JS backend as example.
